deluge/common.py

```diff
@@ -974,6 +974,66 @@ def is_ipv6(ip):
     return False
 
 
+def is_net_interface(interface):
+    """A test to see if 'interface' is a valid network interface.
+
+    Args:
+        interface (str): The Interface to test. eg. eth0 linux. GUID on Windows.
+
+    Returns:
+        bool: Whether Interface is valid or not.
+
+    Examples:
+        >>> is_net_interface("eth0")
+        True
+        >>> is_net_interface("{7A30AE62-23ZA-3744-Z844-A5B042524871}")
+        True
+
+    """
+    if windows_check() and not interface.isupper():
+        log.warning(
+            'Interface is not uppercase you might be looking for: %s', interface.upper()
+        )
+        interface = interface.upper()
+    try:
+        import ifaddr
+
+        adapters = ifaddr.get_adapters()
+        interface = interface.encode('utf-8')
+        for adapter in adapters:
+            try:
+                for iface in interface.split(','):
+                    if interface in adapter.name:
+                        log.info('Interface found with ifaddr!')
+                    else:
+                        raise Exception('Interface not found with ifaddr!', interface)
+                return True
+            except Exception as error:
+                log.warning('Interface not found with ifaddr!', error)
+                return False
+    except ImportError:
+        log.warning('Unable to verify interfaces with ifaddr. Module ifaddr missing')
+        if windows_check():
+            import re
+
+            regex = '^{[0-9a-zA-Z]{8}-([0-9a-zA-Z]{4}-){3}[0-9a-zA-Z]{12}}$'
+            try:
+                for iface in interface.split(','):
+                    if re.search(regex, str(iface)):
+                        log.info(
+                            'Interface found with regex! Could not verify real interface'
+                        )
+                    else:
+                        log.warning('Interface not found with regex!')
+                        raise Exception('Interface not valid!')
```

Comment:
the `log.warning` call seem redundant as in the `except Exception` below we print the same line, but with the exception info.